cmd/cluster/etcd_replace.go

```diff
@@ -0,0 +1,183 @@
+	// Get the etcd pods
+	pods, err := clientset.CoreV1().Pods(EtcdNamespaceName).List(context.TODO(), listOptions)
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		for _, c := range pod.Status.ContainerStatuses {
+			if c.State.Waiting != nil && (c.State.Waiting.Reason == "CrashLoopBackOff" || c.State.Waiting.Reason == "Error") {
+				podName := "etcd-" + opts.nodeId
+				if podName != pod.ObjectMeta.Name {
+					return fmt.Errorf("the etcd member seems to be healthy. Please run health-check again")
+				}
+				err := ReplaceEtcdMember(kubeCli, kconfig, clientset, flags, opts, pod.ObjectMeta.Name)
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("none of the etcd members seems to be unhealthy. Please verify once again")
+}
+
+func ReplaceEtcdMember(kubeCli client.Client, kconfig *rest.Config, clientset *kubernetes.Clientset, flags *genericclioptions.ConfigFlags, opts flagOptions, pod string) error {
+	// Remove the etcd member
+	err := removeEtcdMember(kconfig, clientset, pod, opts)
+	if err != nil {
+		return err
+	}
+
+	// Turn off the quorum guard
+	fmt.Println("[INFO] Turning the quorum guard off")
+	err = patchEtcdQuorum(kubeCli, flags, EtcdQuorumTurnOffPatch)
+	if err != nil {
+		return err
+	}
+
+	// Delete the secrets for the unhealthy etcd member
+	err = removeEtcdSecrets(clientset, opts)
+	if err != nil {
+		return err
+	}
+
+	// Force etcd redeployment.
+	fmt.Println("[INFO] Forcing etcd redeployment")
+	timeStamp := time.Now().Format(time.RFC3339Nano)
+	patch := fmt.Sprintf(EtcdForceRedeployPatch, timeStamp)
+	err = patchEtcdQuorum(kubeCli, flags, patch)
+	if err != nil {
+		return err
+	}
+
+	// Turn the quorum guard back
+	fmt.Println("[INFO] Turning the quorum guard back on")
+	err = patchEtcdQuorum(kubeCli, flags, EtcdQuorumTurnOnPatch)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("The etcd member has been successfully replaced. Please verify by running health check after few minutes.")
+	return nil
+}
+
+func removeEtcdMember(kconfig *rest.Config, clientset *kubernetes.Clientset, pod string, opts flagOptions) error {
+	fmt.Println()
+	cmd := "etcdctl member list -w table | grep " + opts.nodeId + " | awk '{ print $2 }'"
+	memberId, err := Etcdctlhealth(kconfig, clientset, cmd, pod)
+	if err != nil {
+		return err
+	}
+	memberId = strings.TrimSpace(memberId)
+	fmt.Printf("[INFO] Replacing pod %s having member id %s.\n", pod, memberId)
+
+	prompt := utils.ConfirmPrompt()
+	if !prompt {
+		return fmt.Errorf("operation cancelled by user")
+	}
+
+	remove_cmd := "etcdctl member remove " + memberId
+	output, err := Etcdctlhealth(kconfig, clientset, remove_cmd, pod)
+	if err != nil {
+		fmt.Println("Could not replace pod. Refer error below")
+		return err
+	}
+	fmt.Println(output)
+	return nil
+}
+
+func patchEtcdQuorum(kubeCli client.Client, flags *genericclioptions.ConfigFlags, patch string) error {
```

Comment:
As this function is being used for more than just patching quorum, it could maybe be better renamed to something more generic.